assayist/processor/container_go_analyzer.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import os
+import subprocess
+
+from assayist.common.models import content, source
+from assayist.processor.base import Analyzer
+from assayist.processor.logging import log
+
+
+class ContainerGoAnalyzer(Analyzer):
+    """Analyze the Go executables in a container image layer."""
+
+    # goversion executable path.
+    GOVERSION = 'goversion'
+
+    # Backvendor executable path.
+    BACKVENDOR = 'backvendor'
+
+    # Output format tempate for backvendor.
+    BACKVENDOR_TEMPLATE = '\t{{.Ver}}\t{{.Repo}}\t{{.Rev}}'
+
+    # Types of identified source modules
+    UPSTREAM = 'upstream'
+    VENDORED = 'embedded_source_locations'
+
+    def run(self):
+        """Start the container Go analyzer."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+        if not self.is_container_build(build_info):
+            log.info(f'Skipping build {build_id} because the build is not a container')
+            return
+
+        # This container's build is assumed to exist since it is
+        # created by the main analyzer.
+        build = content.Build.nodes.get(id_=build_id)
+        source_location = build.source_location
+        if not source_location:
+            log.error('Missing source location for container build')
+            return
+
+        self._process_source_code(source_location)
+        self._claim_go_executables()
+
+    def _process_source_code(self, source_location):
+        """Run backvendor on the source code.
+
+        :param SourceLocation source_location: local source code DB node
+        """
+        # Find the source code.
+        srcdir = os.path.join(self.input_dir, self.SOURCE_DIR)
+        options = ['-template', self.BACKVENDOR_TEMPLATE]
+        cmd = [self.BACKVENDOR] + options + ['.']
```

Comment:
Perhaps we could check that the `backvendor` (and `goversion`) executable is available with `_assert_command()`?

Reply:
Added a commit to rename `_assert_command()` to `assert_command()`, and updated the Go analyzer to use it.